packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -836,29 +841,35 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
       child: IgnorePointer(
         ignoring: !enabled,
         child: Container(
-          decoration: widget.decoration,
-          // The main decoration and the disabled scrim exists separately.
-          child: Container(
-            color: enabled
-                ? null
-                : CupertinoTheme.of(context).brightness == Brightness.light
-                    ? _kDisabledBackground
-                    : CupertinoColors.darkBackgroundGray,
-            child: TextSelectionGestureDetector(
-              onTapDown: _handleTapDown,
-              onForcePressStart: _handleForcePressStarted,
-              onForcePressEnd: _handleForcePressEnded,
-              onSingleTapUp: _handleSingleTapUp,
-              onSingleLongTapStart: _handleSingleLongTapStart,
-              onSingleLongTapMoveUpdate: _handleSingleLongTapMoveUpdate,
-              onSingleLongTapEnd: _handleSingleLongTapEnd,
-              onDoubleTapDown: _handleDoubleTapDown,
-              onDragSelectionStart: _handleMouseDragSelectionStart,
-              onDragSelectionUpdate: _handleMouseDragSelectionUpdate,
-              onDragSelectionEnd: _handleMouseDragSelectionEnd,
-              behavior: HitTestBehavior.translucent,
-              child: _addTextDependentAttachments(paddedEditable, textStyle, placeholderStyle),
-            ),
+          decoration: widget.decoration == null
+              ? BoxDecoration(
+                  color: disabledColor,
+                )
+              : BoxDecoration(
+                  color: disabledColor,
```

Comment:
I think if `widget.color` is not `null` it should still be honored.

Reply:
But if we copy the `widget.decoration.color`, there will be no visual difference between enabled and disabled. Maybe a new parameter for the disabled color is the way to go?

Reply:
Just did a quick test, with `UITextField` if you give it a non-nil background color it will keep using the same color regardless of the state. 
If the user really wants to distinguish enabled and disabled, a simple workaround would be passing in a different `BoxDecoration` depending on whether the text field is enabled.

Reply:
If I understood you correctly, my last commit change it to your described behaviour.